### PR TITLE
Formatting options refactoring - step 2

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi

File: src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs [hunk 1]
```diff
@@ -49,23 +49,26 @@ public CurlyBraceCompletionService()
...
+            if (!options.AutoFormattingOptions.FormatOnCloseBrace)
+            {
+                return null;
+            }
+
             var (formattingChanges, finalCurlyBraceEnd) = await FormatTrackingSpanAsync(
-                braceCompletionContext.Document,
-                braceCompletionContext.OpeningPoint,
-                braceCompletionContext.ClosingPoint,
-                shouldHonorAutoFormattingOnCloseBraceOption: true,
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs:65]: what a wacky piece of code.

File: src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs [hunk 2]
```diff
@@ -237,17 +243,10 @@ private static bool ContainsOnlyWhitespace(SourceText text, int openingPosition,
...
             ImmutableArray<AbstractFormattingRule> braceFormattingIndentationRules,
-            DocumentOptionSet documentOptions,
+            IndentationOptions options,
             CancellationToken cancellationToken)
         {
-            var option = document.Project.Solution.Options.GetOption(BraceCompletionOptions.AutoFormattingOnCloseBrace, document.Project.Language);
-            if (!option && shouldHonorAutoFormattingOnCloseBraceOption)
-            {
-                return (ImmutableArray<TextChange>.Empty, closingPoint);
-            }
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs:249]: yeesh.

File: src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs [hunk 3]
```diff
@@ -336,7 +336,7 @@ private async Task FormatDocumentCreatedFromTemplateAsync(IVsHierarchy hierarchy
             var formattedText = formattedRoot.GetText(unformattedText.Encoding, unformattedText.ChecksumAlgorithm);
 
             // Ensure the line endings are normalized. The formatter doesn't touch everything if it doesn't need to.
-            var targetLineEnding = formattingOptions.Options.GetOption(FormattingOptions.NewLine)!;
+            var targetLineEnding = formattingOptions.GetOption(FormattingOptions2.NewLine)!;
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs:339]: belagh.  we have `2` options here as well :(

File: src/Workspaces/Core/Portable/Formatting/AutoFormattingOptions.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+{
+    /// <summary>
+    /// Solution-wide formatting options.
+    /// </summary>
+    internal readonly record struct AutoFormattingOptions(
+        FormattingOptions.IndentStyle IndentStyle,
+        bool FormatOnReturn,
+        bool FormatOnTyping,
+        bool FormatOnSemicolon,
+        bool FormatOnCloseBrace)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Formatting/AutoFormattingOptions.cs:22]: yikes.  we codified these c# concepts to the agnostic AutoFormattingOptions?  Is that how it was before as well?

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/Core/Portable/Formatting/AutoFormattingOptions.cs (hunk 4 above)
Comment (tmat) [reply src/Workspaces/Core/Portable/Formatting/AutoFormattingOptions.cs:22]: I'll follow up on moving these options to C#, if possible.

Comment (tmat) [reply src/Workspaces/Core/Portable/Formatting/AutoFormattingOptions.cs:22]: https://github.com/dotnet/roslyn/issues/58917

File: src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs (hunk 3 above)
Comment (tmat) [reply src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs:339]: yeah, actually `-2` options are the ones we should be using everywhere except for public API at this point. I hope to eliminate but not sure yet.

**Pull request merged**